private/signer/v4/v4_test.go

```diff
@@ -23,6 +23,7 @@ func buildSigner(serviceName string, region string, signTime time.Time, expireTi
 	req.Header.Add("Content-Type", "application/x-amz-json-1.0")
 	req.Header.Add("Content-Length", string(len(body)))
 	req.Header.Add("X-Amz-Meta-Other-Header", "some-value=!@#$%^&* (+)")
+	req.Header.Add("X-Amz-Meta-Other-Header_With_Underscore", "some-value=!@#$%^&* (+)")
```

Comment:
Added this test and it failed as you expected due to generating the wrong signature.

I've added a commit and tested it again AWS to make sure it did calculate the correct signature.